solver/src/encoding.rs

```diff
@@ -44,16 +44,23 @@ pub fn encode_trade(
 fn order_flags(order: &OrderCreation) -> U256 {
     let mut result = 0u8;
     result |= match order.kind {
-        OrderKind::Sell => 0,
+        OrderKind::Sell => 0b0,
         OrderKind::Buy => 0b1,
     };
-    if order.partially_fillable {
-        result |= 0b10;
-    };
+    result |= (order.partially_fillable as u8) << 1;
+    result |= match order.sell_token_balance {
+        BalanceFrom::Erc20 => 0b00,
+        BalanceFrom::External => 0b10,
+        BalanceFrom::Internal => 0b11,
+    } << 2;
+    result |= match order.buy_token_balance {
+        BalanceTo::Erc20 => 0b0,
+        BalanceTo::Internal => 0b1,
+    } << 4;
```

Comment:
I am a bit confused by the purpose of these shifts. Could we comment here?

Reply:
`<< 4` is to move the 0 or 1 four bits to the left.